actions/setup-nix/README.md

```diff
@@ -1,32 +1,97 @@
-# setup-nix
+# Setup Nix Action
 
 `setup-nix` installs a nix environment and configures it with the specified
-caches
+caches. Support access to the internal S3 Nix cache in read-only mode and
+read/write mode.
 
 ## Configuration
 
+## Inputs
+
+| Name                    | Description                                                                                        | Required | Default     |
+| ----------------------- | -------------------------------------------------------------------------------------------------- | -------- | ----------- |
+| `cache-url`             | Custom Nix cache URL, for example, `'s3://<bucket-name>'` for an S3-backed cache.                  | Yes      | N/A         |
+| `cache-pubkeys`         | Comma-separated list of custom Nix cache public keys.                                              | Yes      | N/A         |
```

Comment:
This is ok for now, but we should be able to improve on this.

Ideally all trusted pubkeys are published to a single repo, and all other repos just use it from there as configuration, vs. manually specifying their own (duplicates across repos).

Reply:
Yeah make sense. Mitchell is also provisioning them as [Org Wide Variables ](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-an-organization) so it's another option too.